src/Simplex_tree/include/gudhi/Simplex_tree/simplex_tree_options.h

```diff
@@ -99,6 +101,58 @@ template <class, class=void> struct Get_simplex_data_type { typedef No_simplex_d
 template <class O>
 struct Get_simplex_data_type<O, std::void_t<typename O::Simplex_data>> { typedef typename O::Simplex_data type; };
 
+/**
+ * @private
+ * @brief Given two filtration values at which a simplex exists, stores in the first value the minimal union of births
+ * generating a lifetime including those two values.
+ * This is the overload for when `Filtration_value` is a native arithmetic type, like double, int etc.
+ * Because the filtration values are totally ordered then, the union is simply the minimum of the two values.
+ *
+ * NaN values are not supported.
+ */
+template <typename Arithmetic_filtration_value,
+          typename = std::enable_if_t<std::is_arithmetic_v<Arithmetic_filtration_value> > >
+bool unify(Arithmetic_filtration_value& f1, Arithmetic_filtration_value f2)
+{
+  if (f1 > f2){
+    f1 = f2;
+    return true;
+  }
+  return false;
+}
+
+/**
+ * @private
+ * @brief Given two filtration values, stores in the first value the greatest common upper bound of the two values.
+ * If a filtration value has value `NaN`, it should be considered as the lowest value possible.
+ * This is the overload for when `Filtration_value` is a native arithmetic type, like double, float, int etc.
+ * Because the filtration values are totally ordered then, the upper bound is always the maximum of the two values.
+ */
+template <typename Arithmetic_filtration_value,
+          typename = std::enable_if_t<std::is_arithmetic_v<Arithmetic_filtration_value> > >
+bool intersect(Arithmetic_filtration_value& f1, Arithmetic_filtration_value f2)
+{
+  if constexpr (std::is_floating_point_v<Arithmetic_filtration_value>) {
+    if (std::isnan(f1)) {
```

Comment:
So, should I change the `if constexpr` condition to `has_quite_NaN` instead to allow NaNs also for other types than native floating types? I used `std::is_floating_point_v` to avoid the problem with Windows refusing to compile `std::isnan` with integers, so we probably need to keep some kind of condition to avoid this problem again.

Reply:
ok for has_quite_NaN.